### PR TITLE
Update DeviceSimulator2

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: iot-hub/Quickstarts/simulated-device-2/Program.cs [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        private static void CheckForConnectionStringArgument(string[] args)
+        {
+            if (args.Any())
+            {
+                try
+                {
+                    var cs = IotHubConnectionStringBuilder.Create(args.First());
+                    s_connectionString = cs.ToString();
+                }
+                catch (Exception) { }
```
Comment (azabbasi) [inline iot-hub/Quickstarts/simulated-device-2/Program.cs:71]: if we are not doing anything in the catch, we probably don't need the try block either.
I personally think that if this throws we should exit with an error.

File: iot-hub/Quickstarts/simulated-device-2/Program.cs [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+        private static async Task Main(string[] args)
+        {
+            Console.WriteLine("IoT Hub Quickstarts #1 - Simulated device.");
+
+            // This sample accepts the device connection string as a parameter, if present
+            CheckForConnectionStringArgument(args);
+
+            // Connect to the IoT hub using the MQTT protocol
+            s_deviceClient = DeviceClient.CreateFromConnectionString(s_connectionString, TransportType.Mqtt);
```
Comment (azabbasi) [inline iot-hub/Quickstarts/simulated-device-2/Program.cs:41]: suggestion: We can call out the TransportType as a private static field so it's easier to find/change by the users.

**Review by azabbasi — APPROVED** [review]
a couple of nit comments.

File: iot-hub/Quickstarts/simulated-device-2/Program.cs (hunk 1 above)
Comment (drwill-ms) [reply iot-hub/Quickstarts/simulated-device-2/Program.cs:71]: The parameter is optional, and I only want to set it if it passes parsing.

**Pull request merged**